plugins/withViroAndroid.ts

```diff
@@ -0,0 +1,218 @@
+import {
+  withPlugins,
+  withDangerousMod,
+  ConfigPlugin,
+  withAndroidManifest,
+  withProjectBuildGradle,
+  withSettingsGradle,
+  ExportedConfigWithProps,
+  withAppBuildGradle,
+} from "@expo/config-plugins";
+import { Manifest } from "@expo/config-plugins/build/android";
+import { GradleProjectFile } from "@expo/config-plugins/build/android/Paths";
+import { ExpoConfig } from "@expo/config-types";
+import fs from "fs";
+import path from "path";
+import { insertLinesHelper } from "./util/insertLinesHelper";
+
+const withBranchAndroid = (config: ExpoConfig, props: any) => {
+  // Directly edit MainApplication.java
+  config = withDangerousMod(config, [
+    "android",
+    async (config) => {
+      const mainApplicationPath = path.join(
+        config.modRequest.platformProjectRoot,
+        "app",
+        "src",
+        "main",
+        "java",
+        ...(config?.android?.package?.split?.(".") || []),
+        "MainApplication.java"
+      );
+
+      const root = config.modRequest.platformProjectRoot;
+      fs.readFile(mainApplicationPath, "utf-8", (err, data) => {
+        data = insertLinesHelper(
+          "import com.viromedia.bridge.ReactViroPackage;",
+          `package ${config?.android?.package};`,
+          data
+        );
+
+        /**
+         * ********************************************************************
+         * Sample app.json with property config
+         * Options: "AR", "GVR", "OVR_MOBILE"
+         *
+         * https://docs.expo.dev/guides/config-plugins/#using-a-plugin-in-your-app
+         * ********************************************************************
+         *
+         * plugins: [
+         *   [
+         *     "@viro-community/react-viro",
+         *     {
+         *       "androidXrMode": "GVR"
+         *     }
+         *   ]
+         * ],
+         *
+         * ********************************************************************
+         * Sample app.json without property config
+         * The default configuration is "AR"
+         * ********************************************************************
+         *
+         * plugins: [ "@viro-community/react-viro" ],
+         *
+         */
+
+        const viroPlugin = config?.plugins?.find(
+          (plugin) =>
+            Array.isArray(plugin) && plugin[0] === "@viro-community/react-viro"
+        );
+        let viroPluginConfig = "AR";
+        if (
+          Array.isArray(viroPlugin) &&
+          ["AR", "GVR", "OVR_MOBILE"].includes(viroPlugin[1]?.androidXrMode)
+        ) {
+          viroPluginConfig = viroPlugin[1]?.androidXrMode;
+        }
```

Comment:
@doranteseduardo @brunokiafuka I've added some code which allows for the GVR/AR/OVR_MOBILE option in app.json.

Reply:
```javascript
        const viroPlugin = config?.plugins?.find(
          (plugin) =>
            Array.isArray(plugin) && plugin[0] === "@viro-community/react-viro"
        );
```

should we throw an error if the plugin is not in app.json? do we need to worry about that?